### PR TITLE
Make `--target` flag accept directories

### LINKED ISSUE #472: Accept directories as `--target` when specifying tests to run
This will make it easier for the user to run tests.

For example, if the user wants to run all files in the `integration_test/features/geolocation` directory, they'll be able to simply do:

```console
$ patrol drive --target integration_test/features/geolocation
```

### PR DESCRIPTION
**Pull request by bartekpacia**
This PR:
- fixes #472

### CODE REVIEW

**Review by PiotrRogulski — CHANGES_REQUESTED** [review]

File: packages/patrol_cli/lib/src/features/drive/test_finder.dart [hunk 1]
```diff
@@ -12,27 +12,47 @@ class TestFinder {
...
     return targets;
   }
 
   /// Recursively searches the `integration_test` directory and returns files
-  /// that end with `_test.dart` as absolute paths.
-  List<String> findAllTests() {
-    return _integrationTestDirectory
+  /// ending with `_test.dart` as absolute paths.
+  List<String> findAllTests({Directory? directory}) {
+    directory ??= _integrationTestDirectory;
```
[the remark this answers is not in the document]
Comment (bartekpacia) [reply packages/patrol_cli/lib/src/features/drive/test_finder.dart:53]: This is a recommended and preferred workaround around this – [learn more here](https://dart-lang.github.io/linter/lints/parameter_assignments.html). Linter is perfectly happy with this.

File: packages/patrol_cli/test/features/drive/test_finder_test.dart [hunk 2]
```diff
@@ -19,34 +21,189 @@ void main() {
...
+          ),
+        ),
+      );
+    });
+
+    test(
+      'throws exception when target directory does not contain any tests',
+      () {
+        fs
+            .directory('integration_test/features/login')
```
[the remark this answers is not in the document]
Comment (bartekpacia) [reply packages/patrol_cli/test/features/drive/test_finder_test.dart:59]: yeah it is, but it's also what `dart format` outputs... will try to improve things but doubt I'll succeed 😅

File: packages/patrol_cli/lib/src/features/drive/test_finder.dart [hunk 3]
```diff
@@ -12,27 +12,47 @@ class TestFinder {
...
+            'target directory $target does not contain any tests',
+          );
+        }
 
-      final isFile = _fs.isFileSync(target);
-      if (!isFile) {
-        throwToolExit('target file $target does not exist');
+        targets
+          ..insertAll(i + 1, newTargets)
+          ..removeAt(i);
```
[the remark this answers is not in the document]
Comment (bartekpacia) [reply packages/patrol_cli/lib/src/features/drive/test_finder.dart]: Nice idea, `n` will be always small so the performance hit is negligible. Thanks!

File: packages/patrol_cli/test/features/drive/test_finder_test.dart (hunk 2 above)
[the remark this answers is not in the document]
Comment (bartekpacia) [reply packages/patrol_cli/test/features/drive/test_finder_test.dart:59]: OK, I managed to fit this into a single line.

File: packages/patrol_cli/lib/src/features/drive/test_finder.dart (hunk 1 above)
[the remark this answers is not in the document]
Comment (PiotrRogulski) [reply packages/patrol_cli/lib/src/features/drive/test_finder.dart:51]: Ok, I was using a different operator :+1:

**Review by shilangyu — APPROVED** [review]

**Pull request merged**